### DISCUSSION TITLE
Solid red light when trying to echo topics

### DISCUSSION THREAD

**Discussion by JCT03** [iRobotEducation/create3_docs#593] [Questions] [answered]
### How are you connecting to your Create 3?

USB Ethernet (ROS 2)

### Computer(s) Model(s) and Operating System(s)

Raspberry Pi 5, Ubuntu 24.04

### Which version of ROS 2 is installed on your computer?

Other

### Which firmware version is installed on your robot?

H.2.6

### Which RMW is your robot running?

FastRTPS

### Does your robot have an assigned namespace? If so, please share.

/tyrael

### Is the robot connected to a network? If so, what is the network type?

no

### Are there multiple Create 3 robots connected to your network?

No

### Is multicast enabled?

No

### What is the Adapter Board's USB/BLE Toggle currently switched to?

USB

### Describe your question.

I am trying to connect my Raspberry Pi5 running Jazzy to my Create 3 running Humble. I am able to view the topic list and, in some cases, echo topics. However, within 0-5 responses, the light turns solid red and I do not receive any more responses.

**Comment by JCT03** [comment]
[Create3Log.txt](https://github.com/user-attachments/files/16020339/Create3Log.txt)

**Comment by alsora** [comment, marked as answer]
The app is crashing
```
Jun 27 18:55:41 Tyrael user.notice create-platform: terminate called after throwing an instance of 'std::length_error'
Jun 27 18:55:41 Tyrael user.notice create-platform:   what():  vector::_M_default_append
Jun 27 18:55:41 Tyrael user.notice create-platform: Aborted
```

You should not try to communicate between different versions of ROS.
If the robot is using Humble, the raspberry pi must use Humble as well.
Different versions of ROS can serialize data in different ways and cause these crashes

**Reply by JCT03** [reply]
I was afraid of that, thank you for confirming!
